macros/geo/haversine_distance.sql

```diff
@@ -3,17 +3,48 @@ This calculates the distance between two sets of latitude and longitude.
 The formula is from the following blog post:
 http://daynebatten.com/2015/09/latitude-longitude-distance-sql/
 
-The arguments should be float type. 
+The arguments should be float type.
 #}
 
-{% macro haversine_distance(lat1,lon1,lat2,lon2) -%}
-    {{ return(adapter.dispatch('haversine_distance', packages = dbt_utils._get_utils_namespaces())(lat1,lon1,lat2,lon2)) }}
+{% macro degrees_to_radians(degrees) -%}
+    acos(-1) * {{degrees}} / 180
+{%- endmacro %}
+
+{% macro haversine_distance(lat1,lon1,lat2,lon2,unit='mi') -%}
+    {{ return(adapter.dispatch('haversine_distance', packages = dbt_utils._get_utils_namespaces())(lat1,lon1,lat2,lon2,unit)) }}
 {% endmacro %}
 
-{% macro default__haversine_distance(lat1,lon1,lat2,lon2) -%}
+{% macro default__haversine_distance(lat1,lon1,lat2,lon2,unit) -%}
+{# vanilla macro is in miles #}
+    {% set conversion_rate = '' %}
```

Comment:
This is a little cleaner, and has some error handling!
```suggestion
{%- if unit == 'mi' %}
   {% set conversion_rate = 1 %}
{% elif unit == 'km' %}
   {% set conversion_rate = 1.60934 %}
{% else %}
    {{ exceptions.raise_compiler_error("unit input must be one of 'mi' or 'km'. Got ' ~ unit) }}
{% endif -%}
```

Reply:
Thanks thats a lot better indeed! Done :)